### PR TITLE
LPS-133797 Test Autom. Vocabularies in audit graph

### PR DESCRIPTION
**Pull request by Luizcarlosqueiroz**
**JIRA Ticket:**
LPS-133797
LPS-115658

**Motivation:**
Test Automation for "Allow marketers to select which vocabularies to show in the audit graph"

### CODE REVIEW

**Comment by Luizcarlosqueiroz** [comment]
@CarlosBrichete

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboardAuditGraph.testcase [hunk 1]
```diff
@@ -588,72 +588,80 @@ definition {
...
+				category = "Categorization",
+				panel = "Site Administration",
+				portlet = "Categories");
 
-		Vocabulary.addCP(vocabularyName = "Vocabulary Name 1");
+			Vocabulary.addCP(vocabularyName = "Vocabulary Name 1");
 
-		Vocabulary.addCP(vocabularyName = "Vocabulary Name 2");
+			Vocabulary.addCP(vocabularyName = "Vocabulary Name 2");
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboardAuditGraph.testcase:603]: Can you use a JSON macro to create these vocabularies?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboardAuditGraph.testcase [hunk 2]
```diff
@@ -588,72 +588,80 @@ definition {
...
+			ApplicationsMenu.gotoPortlet(
+				category = "Content",
+				panel = "Applications",
+				portlet = "Content Dashboard");
 
-		Click(locator1 = "ContentDashboard#AUDIT_GRAPH_CONFIGURATION_BUTTON");
+			Click(locator1 = "ContentDashboard#AUDIT_GRAPH_CONFIGURATION_BUTTON");
 
-		SelectFrame(locator1 = "IFrame#MODAL_BODY");
+			SelectFrame(locator1 = "IFrame#MODAL_BODY");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboardAuditGraph.testcase:613]: Having a SelectFrame in the .testcase file is a little bit weird... Try creating a macro to go to the Audit Graph configuration :)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboardAuditGraph.testcase [hunk 3]
```diff
@@ -588,72 +588,80 @@ definition {
...
+			Click(locator1 = "ContentDashboard#AUDIT_GRAPH_CONFIGURATION_BUTTON");
 
-		SelectFrame(locator1 = "IFrame#MODAL_BODY");
-
-		Click(
-			columnName = "In Use",
-			locator1 = "ContentDashboard#AUDIT_GRAPH_CONFIGURATION_VOCABULARY_NAME",
-			vocabularyName = "Audience");
+			SelectFrame(locator1 = "IFrame#MODAL_BODY");
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboardAuditGraph.testcase:639]: Having a SelectFrame in the .testcase file is a little bit weird... Try creating a macro to go to the Audit Graph configuration :)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboardAuditGraph.testcase [hunk 4]
```diff
@@ -588,72 +588,80 @@ definition {
...
+			Click(locator1 = "ContentDashboard#AUDIT_GRAPH_CONFIGURATION_LEFT_ARROW_BUTTON");
 
-		Click(
-			columnName = "Available",
-			locator1 = "ContentDashboard#AUDIT_GRAPH_CONFIGURATION_VOCABULARY_NAME",
-			vocabularyName = "Topic");
+			AssertClick(
+				locator1 = "Button#SAVE",
+				value1 = "Save");
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboardAuditGraph.testcase:652]: The move vocabulary from in use to available should be a macro, having clicks in the .testcase file is not very useful

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboardAuditGraph.testcase [hunk 5]
```diff
@@ -588,72 +588,80 @@ definition {
...
+		task ("Move vocabulary from available to in use") {
+			Click(
+				columnName = "Available",
+				locator1 = "ContentDashboard#AUDIT_GRAPH_CONFIGURATION_VOCABULARY_NAME",
+				vocabularyName = "Topic");
 
-		AssertClick(
-			locator1 = "Button#SAVE",
-			value1 = "Save");
+			Click(locator1 = "ContentDashboard#AUDIT_GRAPH_CONFIGURATION_RIGHT_ARROW_BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboardAuditGraph.testcase:660]: Same for this, maybe we can have a macro to move from available to in use.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboardAuditGraph.testcase (hunk 1 above)
Comment (Luizcarlosqueiroz) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboardAuditGraph.testcase:603]: Done!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboardAuditGraph.testcase (hunk 5 above)
Comment (Luizcarlosqueiroz) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboardAuditGraph.testcase:660]: Sure! I've created a macro, but later thought that was unnecessary. But now I'll use it.

**Comment by CarlosBrichete** [comment]
ci:test:tango

**Comment by CarlosBrichete** [comment]
ci:forward

**Pull request closed without merging**